libgalois/src/analytics/pagerank/pagerank-pull.cpp

```diff
@@ -228,26 +224,54 @@ ComputePRTopological(Graph* graph, katana::analytics::PagerankPlan plan) {
 
 }  // namespace
 
+katana::Result<void>
+ExtractPagerank(
+    katana::PropertyGraph* pg, const Graph& from,
+    const std::string& output_property_name) {
+  if (auto result =
+          katana::analytics::ConstructNodeProperties<std::tuple<NodeValue>>(
+              pg, {output_property_name});
+      !result) {
+    return result.error();
+  }
+
+  auto graph_result =
+      katana::TypedPropertyGraph<std::tuple<NodeValue>, std::tuple<>>::Make(
+          pg, {output_property_name}, {});
+  if (!graph_result) {
+    return graph_result.error();
+  }
+  auto graph = graph_result.value();
+
+  katana::do_all(
+      katana::iterate(from),
+      [&](uint32_t i) {
+        PRTy rank = from.GetData<NodeValueOut>(i).value;
+        graph.GetData<NodeValue>(i) = rank;
+      },
+      katana::loopname("Extract pagerank"), katana::no_stats());
+
+  return katana::ResultSuccess();
+}
+
 katana::Result<void>
 PagerankPullTopological(
     katana::PropertyGraph* pg, const std::string& output_property_name,
     katana::analytics::PagerankPlan plan) {
   katana::EnsurePreallocated(2, 3 * pg->num_nodes() * sizeof(NodeData));
-
-  katana::analytics::TemporaryPropertyGuard temporary_property{pg};
-
+  auto compute_graph = std::move(pg->Copy({}, {}).value());
```

Comment:
Is this copying the edges as well? Can we avoid that? @arthurp

Reply:
This is copying edges (and nodes, just not properties). I don't think you need this copy at all. You should just be able to create a temporary property of the correct type and use it. `TemporaryPropertyGuard` will delete the temporary property when you are done. You don't need to copy the graph unless you are sorting/relabeling the edges or nodes.

Reply:
Thank you @arthurp . I now get more how properties and topology are stored and handled. I will fix this. It would be appreciated if I could get your evaluation later.

Reply:
Of course. Though I will be out of the office Thurs and Fri. So I may not be able to do much. But I'll try to check if I can, and if not I'm sure someone else can take a look at this. You may find it useful to look at some of the other algorithms that use intermediate/internal properties while they execute. Several do.